sound/pci/hda/patch_hdmi.c

```diff
@@ -1642,24 +1646,59 @@ static void hdmi_present_sense_via_verbs(struct hdmi_spec_per_pin *per_pin,
 	snd_hda_power_down_pm(codec);
 }
 
+#define SILENT_STREAM_RATE		48000
+#define SILENT_STREAM_CHANNELS		2
+#define SILENT_STREAM_FORMAT		SNDRV_PCM_FORMAT_S16_LE
+#define SILENT_STREAM_MSBITS		16
+
 static void silent_stream_enable(struct hda_codec *codec,
-				struct hdmi_spec_per_pin *per_pin)
+				 struct hdmi_spec_per_pin *per_pin)
 {
-	unsigned int newval, oldval;
+	struct hdmi_spec_per_cvt *per_cvt = NULL;
+	struct hdmi_spec *spec = codec->spec;
+	unsigned int format;
+	int cvt_idx, err;
 
-	codec_dbg(codec, "hdmi: enabling silent stream for NID %d\n",
-			per_pin->pin_nid);
+	codec_dbg(codec, "hdmi: enabling silent stream NID=0x%x\n",
+		  per_pin->pin_nid);
 
 	mutex_lock(&per_pin->lock);
 
 	if (!per_pin->channels)
 		per_pin->channels = 2;
 
-	oldval = snd_hda_codec_read(codec, per_pin->pin_nid, 0,
-			AC_VERB_GET_CONV, 0);
-	newval = (oldval & 0xF0) | 0xF;
-	snd_hda_codec_write(codec, per_pin->pin_nid, 0,
-			AC_VERB_SET_CHANNEL_STREAMID, newval);
+	err = hdmi_choose_cvt(codec, 0, &cvt_idx);
+	if (err) {
+		codec_err(codec, "hdmi: no free converter to enable silent mode\n");
+		return;
+	}
+
+	per_cvt = get_cvt(spec, cvt_idx);
+	/*
+	 * TODO: we need to track cvt assignments for each pin separately
+	 *       and free the converter when silent streaming is ended
+	 */
+	per_cvt->assigned = 0;
+	per_pin->cvt_nid = per_cvt->cvt_nid;
+
+	codec_dbg(codec, "hdmi: using cvt %d (NID=0x%x) for pin NID=0x%x (dev=%d)\n",
+		  cvt_idx, per_cvt->cvt_nid, per_pin->pin_nid, per_pin->dev_id);
+
+	snd_hda_set_dev_select(codec, per_pin->pin_nid, per_pin->dev_id);
```

Comment:
check the return value requried?

Reply:
It seems majority of the calls to set_dev_select are without error checking, so following the same convention here. Not at least for this patch to address.